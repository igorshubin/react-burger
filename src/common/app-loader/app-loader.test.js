import React from 'react';
import renderer from 'react-test-renderer';

import AppLoader from "./app-loader";

it('Loader status: [loading]', () => {
  const statusLoading = renderer
      .create(<AppLoader status={'loading'} />)
      .toJSON();
  expect(statusLoading).toMatchSnapshot();
});

it('Loader status: [error]', () => {
  const statusError = renderer
      .create(<AppLoader status={'error'} />)
      .toJSON();
  expect(statusError).toMatchSnapshot();
});

it('Loader status: [idle]', () => {
  const statusIdle = renderer
      .create(<AppLoader status={'idle'} />)
      .toJSON();
  expect(statusIdle).toMatchSnapshot();
});
