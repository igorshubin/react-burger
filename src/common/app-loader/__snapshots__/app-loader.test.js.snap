// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader status: [error] 1`] = `
<div
  className="app-error"
>
  Ошибка получения данных!
  <button
    className="button button_type_secondary button_size_medium"
    onClick={[Function]}
    type="reset"
  >
    Перегрузить страницу
  </button>
</div>
`;

exports[`Loader status: [idle] 1`] = `null`;

exports[`Loader status: [loading] 1`] = `
<div
  className="app-loading"
>
  Загружаем данные...
</div>
`;
